src/coreComponents/physicsSolvers/SolverBase.cpp

```diff
@@ -458,162 +562,223 @@ real64 SolverBase::nonlinearImplicitStep( real64 const & time_n,
     if( dtAttempt > 0 )
     {
       resetStateToBeginningOfStep( domain );
+      resetConfigurationToBeginningOfStep( domain );
```

Comment:
Why don't you call `resetConfigurationToBeginningOfStep` from the already existing `resetStateToBeginningOfStep`?
There seems to be no need to add another method for `SolverBase`?

Reply:
I think in some cases we want to reset the state without resetting the configuration.